### ISSUE TITLE
startWith causes stream to end early

### ISSUE THREAD

**Issue by AgentME** [baconjs/bacon.js#557]
When you run this:

``` javascript
var Bacon = require('baconjs');
var expandedStateBus = new Bacon.Bus();
expandedStateBus
  .startWith(false)
  .take(1)
  .log('take 1')
  .flatMap(function() {
    return Bacon.later(0, [
      {title: '1'},
      {title: '2'}
    ]);
  })
  .log('last map');
```

you'll get this incorrect output:

```
take 1 false
take 1 <end>
last map <end>
```

If you change `.startWith` to be `.toProperty`, then it works correctly:

```
take 1 false
take 1 <end>
last map [ { title: '1' }, { title: '2' } ]
last map <end>
```

(In case it looks a bit nonsense here: in the original code, `.startWith` isn't always passed false, but an expression that's only sometimes false.)

(Removing the first `.log` statement also fixes it in this stripped-down example. This is not true of the original code I found this issue in.)

Using Bacon 0.7.50 from npm.

**Comment by raimohanska** [comment]
This is practically the same issue as #367: the startWith stream is synchronous and emits it's (only) value to the first subscriber only. The toProperty solution will work of course.

This should be addressed in rel 0.8.

**Comment by raimohanska** [comment]
Fixed in 2.0.
